script.js

```diff
@@ -1,12 +1,1882 @@
+        },
+      },
+    },
+    {
+      id: 1623967,
+      url: "http://www.tvmaze.com/episodes/1623967/game-of-thrones-8x05-the-bells",
+      name: "The Bells",
+      season: 8,
+      number: 5,
+      airdate: "2019-05-12",
+      airtime: "21:00",
+      airstamp: "2019-05-13T01:00:00+00:00",
+      runtime: 79,
+      image: {
+        medium:
+          "http://static.tvmaze.com/uploads/images/medium_landscape/196/491994.jpg",
+        original:
+          "http://static.tvmaze.com/uploads/images/original_untouched/196/491994.jpg",
+      },
+      summary:
+        "<p>Varys betrays his queen, and Daenerys brings her forces to King's Landing.</p>",
+      _links: {
+        self: {
+          href: "http://api.tvmaze.com/episodes/1623967",
+        },
+      },
+    },
+    {
+      id: 1623968,
+      url: "http://www.tvmaze.com/episodes/1623968/game-of-thrones-8x06-the-iron-throne",
+      name: "The Iron Throne",
+      season: 8,
+      number: 6,
+      airdate: "2019-05-19",
+      airtime: "21:00",
+      airstamp: "2019-05-20T01:00:00+00:00",
+      runtime: 80,
+      image: {
+        medium:
+          "http://static.tvmaze.com/uploads/images/medium_landscape/198/495648.jpg",
+        original:
+          "http://static.tvmaze.com/uploads/images/original_untouched/198/495648.jpg",
+      },
+      summary:
+        "<p>In the aftermath of the devastating attack on King's Landing, Daenerys must face the survivors.</p>",
+      _links: {
+        self: {
+          href: "http://api.tvmaze.com/episodes/1623968",
+        },
+      },
+    },
+  ];
+}
+
+let getEpi = getAllEpisodes();
+
+function getEpiName(getEpi) {
+  for (let i = 0; i < getEpi.length; i++) {
+    let option = document.createElement("option");
+    option.text = `Seasont${getEpi[i].season} Episode${getEpi[i].number} : ${getEpi[i].name}`;
+    episodeSelect.appendChild(option);
+  }
+}
+getEpiName(getEpi);
+
+function selectEpisodes(episodeSelect) {
+  const episodes = getAllEpisodes();
+  episodeSelect.innerHTML = "";
+
+  const defaultOption = document.createElement("option");
+  defaultOption.textContent = "Please select...";
+  episodeSelect.appendChild(defaultOption);
+
+  episodes.forEach((episode) => {
+    const optionElement = document.createElement("option");
+    optionElement.textContent = episode.name;
+    episodeSelect.appendChild(optionElement);
+  });
+
+  episodeSelect.addEventListener("change", (event) => {
+    const selectedEpisodeName = event.target.value;
+
+    if (selectedEpisodeName === "Please select...") {
+      clearRender();
+      const allEpisodes = getAllEpisodes();
+      makePageForEpisodes(allEpisodes);
+    } else {
+      const filteredEpisodes = episodes.filter(
+        (episode) => episode.name === selectedEpisodeName
+      );
+      clearRender();
+      makePageForEpisodes(filteredEpisodes);
+    }
+  });
+}
+
+episodeSelector.addEventListener("input", function () {
```

Comment:
Do you think we should use "input" or "change" as the event listener? Look up what the difference is between them